test/fuzzy_object_store/fuzz_object.cpp

```diff
@@ -0,0 +1,569 @@
+        for (auto col : all_col_keys) {
+            StringData col_name = t->get_column_name(col);
+            static_cast<void>(col_name);
+        }
+    }
+}
+
+void FuzzObject::commit(SharedRealm shared_realm, FuzzLog& log)
+{
+    log << "FuzzObject::commit();\n";
+    log << "FuzzObject::commit() - shared_realm->is_in_transaction();\n";
+    if (shared_realm->is_in_transaction()) {
+        log << "FuzzObject::commit() - shared_realm->commit_transaction();\n";
+        shared_realm->commit_transaction();
+        auto& group = shared_realm->read_group();
+        REALM_DO_IF_VERIFY(log, group.verify());
+    }
+}
+
+void FuzzObject::rollback(SharedRealm shared_realm, Group& group, FuzzLog& log)
+{
+    log << "FuzzObject::rollback()\n";
+    if (!shared_realm->is_in_async_transaction() && !shared_realm->is_in_transaction()) {
+        shared_realm->begin_transaction();
+        REALM_DO_IF_VERIFY(log, group.verify());
+        log << "shared_realm->cancel_transaction();\n";
+        shared_realm->cancel_transaction();
+        REALM_DO_IF_VERIFY(log, shared_realm->read_group().verify());
+    }
+}
+
+void FuzzObject::advance(realm::SharedRealm shared_realm, FuzzLog& log)
+{
+    log << "FuzzObject::advance();\n";
+    shared_realm->notify();
+}
+
+void FuzzObject::close_and_reopen(SharedRealm& shared_realm, FuzzLog& log, const Realm::Config& config)
+{
+    log << "Open/close realm\n";
+    shared_realm->close();
+    shared_realm.reset();
+    shared_realm = Realm::get_shared_realm(config);
+    log << "Verify group after realm got reopened\n";
+    auto& group = shared_realm->read_group();
+    REALM_DO_IF_VERIFY(log, group.verify());
+}
+
+void FuzzObject::create_table_view(Group& group, FuzzLog& log, State& s, std::vector<TableView>& table_views)
+{
+    log << "FuzzObject::create_table_view();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    log << "table_views.push_back(wt->get_table(" << table_key << ")->where().find_all());\n";
+    TableView tv = t->where().find_all();
+    table_views.push_back(tv);
+}
+
+void FuzzObject::check_null(Group& group, FuzzLog& log, State& s)
+{
+    log << "FuzzObject::check_null();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    if (t->get_column_count() > 0 && t->size() > 0) {
+        auto all_col_keys = t->get_column_keys();
+        size_t ndx = get_next_token(s) % all_col_keys.size();
+        ColKey col = all_col_keys[ndx];
+        ObjKey key = t->get_object(get_int32(s) % t->size()).get_key();
+        log << "group.get_table(" << table_key << ")->get_object(" << key << ").is_null(" << col << ");\n";
+        bool res = t->get_object(key).is_null(col);
+        static_cast<void>(res);
+    }
+}
+
+void FuzzObject::async_write(SharedRealm shared_realm, FuzzLog& log)
```

Comment:
If async transactions are to be tested then we must have a runloop in the bottom of the main thread. Compare how it is done in the branch `je/verify_cluster`.

Reply:
I am going to drop support for async transactions for now, we can only add them back. The goal of this PR is to get the framework up and running and then start adding into it. I can have an issue to track this.